File: app/java/src/main/java/com/example/production/domain/model/quality/InspectionJudgment.java
package com.example.production.domain.model.quality;

/**
 * 検査判定
 */
public enum InspectionJudgment {
    PASSED("合格"),
    FAILED("不合格"),
    HOLD("保留");

    private final String displayName;

    InspectionJudgment(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }

    public static InspectionJudgment fromDisplayName(String displayName) {
        for (InspectionJudgment judgment : values()) {
            if (judgment.displayName.equals(displayName)) {
                return judgment;
            }
        }
        throw new IllegalArgumentException("Unknown display name: " + displayName);
    }
}

File: app/csharp/src/ProductionManagement.Domain/Models/Quality/InspectionJudgment.cs
namespace ProductionManagement.Domain.Models.Quality;

/// <summary>
/// 検査判定
/// </summary>
public enum InspectionJudgment
{
    Passed,
    Failed,
    Hold
}

/// <summary>
/// 検査判定の拡張メソッド
/// </summary>
public static class InspectionJudgmentExtensions
{
    private static readonly Dictionary<InspectionJudgment, string> DisplayNames = new()
    {
        { InspectionJudgment.Passed, "合格" },
        { InspectionJudgment.Failed, "不合格" },
        { InspectionJudgment.Hold, "保留" }
    };

    public static string ToDisplayName(this InspectionJudgment judgment)
        => DisplayNames[judgment];

    public static InspectionJudgment FromDisplayName(string displayName)
        => DisplayNames.First(x => x.Value == displayName).Key;
}
